homeassistant/components/senseme/config_flow.py

```diff
@@ -0,0 +1,135 @@
+"""Config flow for SenseME."""
+from __future__ import annotations
+
+import ipaddress
+from typing import Any
+
+from aiosenseme import SensemeDevice, async_get_device_by_ip_address
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE, CONF_HOST, CONF_ID
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.typing import DiscoveryInfoType
+
+from .const import CONF_HOST_MANUAL, CONF_INFO, DOMAIN
+from .discovery import async_discover, async_get_discovered_device
+
+DISCOVER_TIMEOUT = 5
+
+
+class SensemeFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle SenseME discovery config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the SenseME config flow."""
+        self._discovered_devices: list[SensemeDevice] | None = None
+        self._discovered_device: SensemeDevice | None = None
+
+    async def async_step_discovery(
+        self, discovery_info: DiscoveryInfoType
+    ) -> FlowResult:
+        """Handle discovery."""
+        uuid = discovery_info[CONF_ID]
+        device = async_get_discovered_device(self.hass, discovery_info[CONF_ID])
+        host = device.address
+        await self.async_set_unique_id(uuid)
+        for entry in self._async_current_entries(include_ignore=False):
+            if entry.data[CONF_INFO]["address"] == host:
```

Comment:
`_abort_if_unique_id_configured` doesn't work here because we need to merge the new address into the `CONF_INFO` `dict` and it doesn't support deep merges

Reply:
Can't we pass the whole updated data as updates? We have access to the old and the new data.

Reply:
Sorry we don't have access to the old data. 🤦